opm/parser/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -766,6 +773,36 @@ namespace Opm {
 
         return index(i,j,k,c);
     }
+
+    void ZcornMapper::fixupZCORN( std::vector<double>& zcorn, int sign, size_t i, size_t j, size_t k , size_t c) {
+        size_t index1 = this->index(i,j,k,c);
+        size_t index2 = this->index(i,j,k,c+4);
+
+        if ((zcorn[index2] - zcorn[index1]) * sign < 0 )
+            zcorn[index2] = zcorn[index1];
+
+    }
+
+    void ZcornMapper::fixupZCORN( std::vector<double>& zcorn) {
```

Comment:
I think points might be the most interesting, since that would correspond to the number of "wrong" zcorn values in the input. Although I would expect it to be divisible by 4 quite often...